plugins/module_utils/network/sonic/config/sflow/sflow.py

```diff
@@ -0,0 +1,507 @@
+                                 "port": collector["port"]}
+            # since REST needs the collector list item with its settings and a nested config with a copy of those same settings
+            collector_request.update({"config": dict(collector_request)})
+            collector_list.append(collector_request)
+        return collector_list
+
+    def create_interface_list_request_body(self, config_dict):
+        '''does format transformation and creates and returns a list of sflow interfaces with the settings passed in.
+        Takes a dictionary for all config in argspec format and returns all interfaces listed that have configuration. Returns list in REST API format'''
+        interface_list = []
+        for interface in config_dict["interfaces"]:
+            interface_config_request = {}
+            if "enabled" in interface:
+                interface_config_request["enabled"] = interface["enabled"]
+            if "sampling_rate" in interface:
+                interface_config_request["sampling-rate"] = interface["sampling_rate"]
+            if len(interface_config_request) == 0:
+                # listed interface doesn't actually have any configured settings, but name is hanging around
+                continue
+            interface_config_request["name"] = interface["name"]
+            interface_list.append({"name": interface["name"],
+                                   "config": interface_config_request})
+        return interface_list
+
+    def get_deleted_requests(self, to_delete, have):
+        '''get list of requests needed to delete all settings in to_delete. have is needed to help delete multiple collectors at once and all of an interface'''
+        requests = []
+        if "enabled" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/enabled", "method": "PUT",
+                             "data": {"openconfig-sampling-sflow:enabled": False}})
+        if "polling_interval" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/polling-interval", "method": "DELETE"})
+        if "agent" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/agent", "method": "DELETE"})
+        if "sampling_rate" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/sampling-rate", "method": "DELETE"})
+        if "collectors" in to_delete:
+            have_collectors_dict = {(collector["address"], collector["network_instance"], collector["port"]): collector for collector in have["collectors"]}
+            to_delete_collectors_dict = {(collector["address"], collector["network_instance"], collector["port"]): collector
+                                         for collector in to_delete["collectors"]}
+            if have_collectors_dict.keys() == to_delete_collectors_dict.keys():
```

Comment:
python3 has it so that dict().keys() returned value behaves like a set  https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects Though explicit set creation sounds better for backwards compatibility